webhook/client/client.go

```diff
@@ -33,22 +35,36 @@ func New(ctx context.Context, config *rest.Config, namespace string) (*Client, e
 		return nil, err
 	}
 
-	kubeClient, err := clientset.NewForConfig(config)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to get k8s client")
-	}
+	if serverType == types.WebhookTypeAdmission {
+		kubeClient, err := clientset.NewForConfig(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to get k8s client")
+		}
 
-	lhClient, err := lhclientset.NewForConfig(config)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to get lh client")
-	}
+		lhClient, err := lhclientset.NewForConfig(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to get lh client")
+		}
+
+		kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, time.Second*30)
+		lhInformerFactory := lhinformers.NewSharedInformerFactory(lhClient, time.Second*30)
 
-	kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, time.Second*30)
-	lhInformerFactory := lhinformers.NewSharedInformerFactory(lhClient, time.Second*30)
-	datastore := datastore.NewDataStore(lhInformerFactory, lhClient, kubeInformerFactory, kubeClient, namespace)
+		datastore := datastore.NewDataStore(lhInformerFactory, lhClient, kubeInformerFactory, kubeClient, namespace)
+
+		go kubeInformerFactory.Start(ctx.Done())
+		go lhInformerFactory.Start(ctx.Done())
+
+		if !datastore.Sync(ctx.Done()) {
+			return nil, fmt.Errorf("datastore cache sync up failed")
+		}
+
+		return &Client{
+			Clients:   *clients,
+			Datastore: datastore,
+		}, nil
+	}
 
```

Comment:
NIT: Do we still need to check if `serverType == types.WebhookTypeConversion`?

Reply:
Yes, seem better.